src/components/molecules/ChatList/ChatMessage.tsx

```diff
@@ -55,33 +50,66 @@ export const ChatMessage: FC<ChatMessageProps> = ({
     onDeleteClick(message);
   }, [message, onDeleteClick]);
 
-  return message.from in usersById ? (
+  if (!user || !profile) return null;
+
+  const sender = { ...usersById[message.from], id: message.from };
+  const isMe = message.from === user?.uid;
+
+  const getProfileImage = (): string => {
+    if (isMe) {
+      return profile.pictureUrl || "";
+    }
+
+    if (sender.anonMode) {
+      return DEFAULT_PROFILE_IMAGE;
+    }
+
+    return sender.pictureUrl || "";
+  };
+
+  const getProfileName = (): string => {
+    if (isMe) {
+      return profile.partyName || "";
+    }
+
+    if (sender.anonMode) {
+      return DEFAULT_PARTY_NAME;
+    }
+
+    return sender.partyName || "";
+  };
+
+  return (
     <div
       className={`message chat-message ${isMe ? "chat-message_own" : ""}`}
       key={`${message.from}-${message.ts_utc}`}
     >
       <div className="chat-message-bubble">{getLinkFromText(message.text)}</div>
       <div className="chat-message-author">
-        <img
-          onClick={showMessage}
-          key={`${message.from}-messaging-the-band`}
-          className="chat-message-avatar"
-          src={profileImage}
-          title={profileName}
-          alt={`${profileName} profile`}
-          width={PROFILE_IMAGE_SIZE}
-          height={PROFILE_IMAGE_SIZE}
-        />
-        <div className="chat-message-pseudo">
-          {profileName}{" "}
-          <span className="timestamp">{formatUtcSeconds(message.ts_utc)}</span>
-        </div>
+        {showSenderImage && (
+          <>
+            <img
+              onClick={showMessage}
+              key={`${message.from}-messaging-the-band`}
+              className="chat-message-avatar"
+              src={getProfileImage()}
+              title={getProfileName()}
+              alt={`${getProfileName()} profile`}
+              width={PROFILE_IMAGE_SIZE}
+              height={PROFILE_IMAGE_SIZE}
+            />
+            <div className="chat-message-pseudo">
+              {getProfileName()}{" "}
+              <span className="timestamp">
+                {formatUtcSeconds(message.ts_utc)}
```

Comment:
This will cause a type error once https://github.com/sparkletown/sparkle/pull/897 gets merged.
```suggestion
                {formatUtcSeconds(message.ts_utc.seconds)}
```

Reply:
resolved in https://github.com/sparkletown/sparkle/pull/921/commits/f69bd1a816115991a5ca4257830f370237e55697